### PR TITLE
Neovim support

### PR DESCRIPTION
**Pull request by huyu398**
Original preview-markdown.vim is not support for Neovim; it will occur an error on Neovim.
My pull-request would support Neovim, but I'm not expert of Neovim and vim-script.
Sorry for my code if this PR has bad implementation.

I think this function will have better other implemtation.

### CODE REVIEW

File: autoload/preview_markdown.vim [hunk 1]
```diff
@@ -47,7 +47,12 @@ function! preview_markdown#preview() abort
     let opt['term_finish'] = 'close'
   endif
 
-  call term_start(parser, opt)
+  if has('nvim')
+    vnew
```
Comment (skanehira) [inline autoload/preview_markdown.vim]: If `get(g:, 'preview_markdown_vertical', 0)` return 1, then should execute `vnew`

File: autoload/preview_markdown.vim [hunk 2]
```diff
@@ -47,7 +47,12 @@ function! preview_markdown#preview() abort
...
+    call termopen('mdr ' . tmp, {'on_exit': function('s:remove_tmp', [tmp])})
```
Comment (skanehira) [inline autoload/preview_markdown.vim]: The command should use variable `parser`.

```vim
termopen(printf('%s %s', parser, tmp), {'on_exit': function('s:remove_tmp', [tmp])})
```

**Review by skanehira — COMMENTED** [review]
Thank you for PR :)
I have some comment

File: autoload/preview_markdown.vim (hunk 2 above)
Comment (skanehira) [reply autoload/preview_markdown.vim]: Also, Neovim's `remove_tmp` function args should be this.

```
							*on_exit*
Arguments passed to on_exit callback:
  0: |job-id|
  1: Exit-code of the process, or 128+SIGNUM if by signal (e.g. 143 on SIGTERM).
  2: Event type: "exit"
```

**Comment by skanehira** [comment]
@huyu398 I think it looks like this :)
```diff
diff --git a/autoload/preview_markdown.vim b/autoload/preview_markdown.vim
index ecf08bc..d4321ab 100644
--- a/autoload/preview_markdown.vim
+++ b/autoload/preview_markdown.vim
@@ -13,8 +13,8 @@ endfunction

 function! preview_markdown#preview() abort
   if wordcount().bytes is 0
-      call s:echo_err('current buffer is empty')
-      return
+    call s:echo_err('current buffer is empty')
+    return
   endif

   let tmp = tempname() . '.md'
@@ -31,28 +31,46 @@ function! preview_markdown#preview() abort
     return
   endif

-  if !has('terminal')
+  if !has('terminal') && !has('nvim')
     call s:echo_err('this version doesn''t support terminal')
     return
   endif

-  let opt = {
-        \ 'in_io': 'file',
-        \ 'in_name': tmp,
-        \ 'exit_cb': function('s:remove_tmp', [tmp]),
-        \ 'vertical': get(g:, 'preview_markdown_vertical', 0),
-        \ }
+  if has('nvim')
+    if get(g:, 'preview_markdown_vertical', 0) is 1
+      vnew
+    endif

-  if parser is 'mdr'
-    let opt['term_finish'] = 'close'
-  endif
+    let opt = {
+          \ 'on_exit': function('s:remove_tmp_on_nvim', [tmp])
+          \ }
+
+    let cmd = printf("%s %s", parser, tmp)
+
+    call termopen(cmd, opt)
+  else
+    let opt = {
+          \ 'in_io': 'file',
+          \ 'in_name': tmp,
+          \ 'exit_cb': function('s:remove_tmp', [tmp]),
+          \ 'vertical': get(g:, 'preview_markdown_vertical', 0),
+          \ }
+
+    if parser is 'mdr'
+      let opt['term_finish'] = 'close'
+    endif

-  call term_start(parser, opt)
+    call term_start(parser, opt)
+  endif
 endfunction

 function! s:remove_tmp(tmp, channel, msg) abort
   call delete(a:tmp)
 endfunction

+function! s:remove_tmp_on_nvim(tmp, id, exit_code, type) abort
+  call delete(a:tmp)
+endfunction
+
 let &cpo = s:save_cpo
 unlet s:save_cpo
```

**Comment by huyu398** [comment]
@skanehira
Thank you for your comments!!
I implemented by your comments, please check my code.

File: autoload/preview_markdown.vim [hunk 3]
```diff
@@ -31,28 +31,48 @@ function! preview_markdown#preview() abort
...
-        \ 'in_name': tmp,
-        \ 'exit_cb': function('s:remove_tmp', [tmp]),
-        \ 'vertical': get(g:, 'preview_markdown_vertical', 0),
-        \ }
+  if has('nvim')
+    if get(g:, 'preview_markdown_vertical', 0) is 1
+      vnew
+    else
+      new
+    endif
```
Comment (huyu398) [inline autoload/preview_markdown.vim:44]: `g:preview_markdown_vertical` is not defined by default, so default behavior is horizontal opening.
But, sample image in README.md is vertical opening.
I would suggest `g:preview_markdown_vertical` is 1 by default, or it horizontally open if `g:preview_markdown_horizontal` is 1.
This settings will be intuition for users reading README.md.

File: autoload/preview_markdown.vim [hunk 4]
```diff
@@ -31,28 +31,48 @@ function! preview_markdown#preview() abort
...
-  call term_start(parser, opt)
+    let cmd = printf("%s %s", parser, tmp)
+
+    call termopen(cmd, opt)
+  else
+    let opt = {
+          \ 'in_io': 'file',
+          \ 'in_name': tmp,
+          \ 'exit_cb': function('s:remove_tmp', [tmp]),
+          \ 'vertical': get(g:, 'preview_markdown_vertical', 0),
```
Comment (huyu398) [inline autoload/preview_markdown.vim:58]: this same too.

File: autoload/preview_markdown.vim (hunk 3 above)
Comment (skanehira) [reply autoload/preview_markdown.vim:44]: I don't want to add too many options,
so I think better that vertical when preview_markdown_vertical is 1 and horizontal when undefined.

**Review by skanehira — APPROVED** [review]
LGTM :)
Thank you for PR :+1:

**Pull request merged**